bouncer/embed_detector.py

```diff
@@ -0,0 +1,45 @@
+import fnmatch
+import re
+from urllib.parse import urlparse
+
+"""
+Hardcoded URL patterns where client is assumed to be embedded.
+
+Only the hostname and path are included in the pattern. The path must be
+specified; use "example.com/*" to match all URLs on a particular domain.
+
+Patterns are shell-style wildcards ('*' matches any number of chars, '?'
+matches a single char).
+"""
+PATTERNS = [
+    "h.readthedocs.io/*",
+    "web.hypothes.is/blog/*",
+]
+
+COMPILED_PATTERNS = [re.compile(fnmatch.translate(pat)) for pat in PATTERNS]
```

Comment:
I'm not super familiar but I'm wondering why it's necessary to translate the fnmatch pattern into a regex? Can't fnmatch be used directly to match against the URL?

Reply:
`fnmatch` internally works by translating the shell-style wildcard into a regex which is then matched against the target string. This is a small optimization by pre-compiling the target pattern.